File: back/src/test/java/com/tdevilleduc/urthehero/back/model/PersonTest.java
package com.tdevilleduc.urthehero.back.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils;

public class PersonTest {

    @Test
    public void test_Constructor() {
        String personLogin = RandomStringUtils.random(20);
        String personDisplayName = RandomStringUtils.random(20);
        String personEmail = RandomStringUtils.random(20);
        Person person = new Person(personLogin, personDisplayName, personEmail);

        Assertions.assertEquals(person.getLogin(), personLogin);
        Assertions.assertEquals(person.getDisplayName(), personDisplayName);
        Assertions.assertEquals(person.getEmail(), personEmail);

        Person secondPerson = new Person();
        secondPerson.setLogin(personLogin);
        secondPerson.setDisplayName(personDisplayName);
        secondPerson.setEmail(personEmail);

        Assertions.assertEquals(secondPerson.toString(), person.toString());
        Assertions.assertEquals(secondPerson, person);
    }
}

File: back/src/test/java/com/tdevilleduc/urthehero/back/model/PersonTest.kt
package com.tdevilleduc.urthehero.back.model

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.apache.commons.lang3.RandomStringUtils

internal class PersonTest {
    @Test
    fun test_defaultRole_isRoleUser() {
        val user = User(null, "username", "password")
        Assertions.assertEquals("ROLE_USER", user.getRole())
    }

    @Test
    fun test_adminRole() {
        val user = User(null, "username", "password", "ROLE_ADMIN")
        Assertions.assertEquals("ROLE_ADMIN", user.getRole())
        Assertions.assertEquals(1, user.authorities.size)
        Assertions.assertEquals("ROLE_ADMIN", user.authorities.first().authority)
    }

    @Test
    fun test_setRole() {
        val user = User(null, "username", "password")
        user.setRole("ROLE_ADMIN")
        Assertions.assertEquals("ROLE_ADMIN", user.getRole())
    }
}
